### DISCUSSION TITLE
Nested for each and new lines

### DISCUSSION THREAD

**Discussion by MarkGTI** [rfennell/AzurePipelines#1275] [Q&A]
So I got most of this working

{{#each_with_sort_by_field  this.workItems "Custom.ReleaseNotesSection"}}
{{lookup this.fields 'Custom.ReleaseNotesSection'}}
{{this.id}}  {{lookup this.fields 'Custom.ReleaseNotesTitle'}} :{{{lookup this.fields 'Custom.ReleaseNotes'}}}
{{/each_with_sort_by_field}}

So this gets all work items sorted by Custom.ReleaseNotesSection
 So should be like 

AP
Bug 1
Bug 2
Bug 2

AR
Bug 4
Bug 5

Problem is I'm getting this:

**AP**
Bug 1
**AP**
Bug 2
**AP**
Bug 2

**AR**
Bug 4
**AR**
Bug 5

I only need the  heading the first time it occurs. normally I'd do something like store the current heading and then have an if clause that only displays it if it's different OR nested loops, get all work items sorted and if first then have another loop where ReleaseNotesSection is same as current heading

Anyone got any ideas how to do this with handlebars

**Comment by rfennell** [comment]
You use the {{#if isFirst}} construct inside the loop e.g

{{#forEach pullRequests}} 
{{#if isFirst}}### Pull Requests{{/if}}

- {{this.pullRequestId}} {{this.title}}

 {{/forEach}}

**Reply by MarkGTI** [reply]
thanks, I had tried that, but it only shows the first ReleaseNotesSection. I need to show it when it changes

**Reply by rfennell** [reply]
I think it would need a custom extension. This Stackoverflow issue seems to be in the right direction

https://stackoverflow.com/questions/49268974/accessing-previous-index-value-while-looping-with-each-in-handlebars
